runtime/lua/vim/shared.lua

```diff
@@ -497,7 +497,7 @@ vim.validate = (function()
     return tname
   end
   local function _is_type(val, t)
-    return t == 'callable' and vim.is_callable(val) or type(val) == t
+    return type(val) == t or (t == 'callable' and vim.is_callable(val))
```

Comment:
This is just a performance optimization for mostly no reason. Callable is not as common as the other types.